### ISSUE TITLE
[2.5] Cannot translate some English phrases

### ISSUE THREAD

**Issue by gn1github** [bitshares/bitshares-ui#1243]
After #1186 was added, Japanese version became available on frontend. I am currently trying to fix up the phrase to match the context.

I have realized that there are strings that are still in English, but I've realized that the string itself is not included in the original EN JSON file.

Just as an example, there is a phrase "Get 80% Cashback on Fees" and "Lifetime Members get 80% cashback on every transaction fee they pay and qualify to earn referral income from users they register with or refer to the network. A Lifetime Membership is just 110.80263 BTS." in the membership stats page as shown below.

<img width="881" alt="membership-stats-page" src="https://user-images.githubusercontent.com/36555761/36902571-4d2c8f1c-1e6e-11e8-9031-f9514faa541b.png">

These strings do not exist in the original English JSON file, so I am not sure how to translate such phrases to Japanese. (And there are many of these.) Kindly advise on how to integrate it.

**Comment by startailcoon** [comment]
@wmbutler I can take some time to fix all the hardcoded English to translation tags.

**Comment by startailcoon** [comment]
The sections you're looking for are located in `app/help` as .md files.
These are called by the function HelpContent() in `app/components/Utility/HelperContent.jsx`.

There are still a few items in the app that are missing <Translation> items for other languages though.

**Comment by gn1github** [comment]
I am not sure how to use the information you gave me about the md files, but are you able to convert the hardcoded English into translatable strings and add them onto locale-en.json file?

**Comment by startailcoon** [comment]
The items you where unable to translate are not part of the hard-coded strings I mentioned, even though I found some that needs fixing.

What you do is this:
- Take the files located in https://github.com/bitshares/bitshares-ui/tree/staging/app/help/en
- Create a new folder called "jp" in the /help folder
- Put all the files in the new folder
- Translate the required files

The section you're unable to find is located in 
https://github.com/bitshares/bitshares-ui/blob/staging/app/help/en/components/AccountMembership.md

You can see how the other languages are made, like Russian that's pretty comprehensively made
https://github.com/bitshares/bitshares-ui/tree/staging/app/help/ru

**Comment by gn1github** [comment]
Thank you for pointing me to the right direction.

I have taken AccountMembership.md from the en folder, translated into Japanese, and proposed to create folders & file called ja/components/AccountMembership.md in the /help folder. (Seems like I don't have permission to upload files directly)

Please confirm the proposal above.

**Comment by wmbutler** [comment]
You need to:

- for the repo to your own account
- create a branch from staging `git checkout -b 1243` (1243 is this issue)
- make your changes to that branch
- commit it and push it back up to your repo
- issue a Pull Request back to this repo of that 1243 branch

**Comment by gn1github** [comment]
I followed your instructions, and #1266 was created.

So, every time I create a new translation file, I should follow above procedure and open a pull request, correct?

**Comment by gn1github** [comment]
@startailcoon Please see below screenshot from the "Borrow bitUSD" screen. There are English phrases (explanation and the button) that need to be translated to Japanese.

<img width="698" alt="borrowbitusdscreen" src="https://user-images.githubusercontent.com/36555761/37030202-72b4db2e-217d-11e8-98d0-61613f89521d.png">

Are these part of the hardcoded English phrases that you were talking about, or are these also available in a .md file in one of the folders that wasn't mentioned here?

**Comment by wmbutler** [comment]
@gn1github yes, PR's get assigned a number as well. Typically in your commit message, you would reference this original issue number like "#1243" You cab also manually link it on the PR.

**Comment by startailcoon** [comment]
> @gn1github Are these part of the hardcoded English phrases that you were talking about, or are these also available in a .md file in one of the folders that wasn't mentioned here?

Yes, that is one of the hard coded English parts. It's addressed in another PR from one of the developers and will soon be fixed when PR #1191 is merged.

**Comment by startailcoon** [comment]
Thanks for all your findings. This is great, and I'll fix these spots. Just post here with all you find and we'll iron them out.

**Comment by startailcoon** [comment]
# A few solutions and issues

### Issue
The issue I have is that I can't find a way to make a `<Translation>` tag for something that has another tag for usage, like `<AssetName>` or `<Link>` for example

### Login Screen
The login screen can be fixed by redoing it to something like this, but needs to be accepted by the UI team.

![image](https://user-images.githubusercontent.com/12114550/37113505-dd6bf6f0-2245-11e8-9e2d-0e7366cf75fd.png)

### Market Buy, Deposit, Buy Links
The Deposit and Buy links can be fixed, but the Buy text contains the `<AssetName>` tag, which is an issue that I need some help with finding a solution for.

### Market "star only"
This is possible for you as the developers have solved this by having two translations on each side, as it's made like this.

`exchange.show_star_1` (star) `exchange.show_star_2`

**Comment by svk31** [comment]
Check the TranslateWithLinks component, it does what you're looking for. It's used heavily in Operation.jsx for example: https://github.com/bitshares/bitshares-ui/blob/staging/app/components/Blockchain/Operation.jsx#L536

**Comment by startailcoon** [comment]
@gn1github sorry, I misunderstood your issue with the translation by the star icon.

**Comment by wmbutler** [comment]
Has anyone claimed this?

**Comment by gn1github** [comment]
@startailcoon No worries. Would you be able to add the ability to reorder the star icon?

**Comment by startailcoon** [comment]
> @wmbutler Has anyone claimed this?

I have it assigned to me and are working on it.

**Comment by startailcoon** [comment]
Once PR #1273 is merged, the translation file will be updated to support the issues related here.

Login Box is changeable at
https://github.com/bitshares/bitshares-ui/blob/978bd1a4c73e312b7cfd71444d4ac3bf5ff4affc/app/assets/locales/locale-en.json#L170-L174

Buy/Sell Box is changeable at
https://github.com/bitshares/bitshares-ui/blob/978bd1a4c73e312b7cfd71444d4ac3bf5ff4affc/app/assets/locales/locale-en.json#L1152

Star icon is changeable at
https://github.com/bitshares/bitshares-ui/blob/978bd1a4c73e312b7cfd71444d4ac3bf5ff4affc/app/assets/locales/locale-en.json#L1252

**Comment by startailcoon** [comment]
Closing Issue as all sections reported needed fixing is now completed.

**Comment by gn1github** [comment]
Please allow me to reopen this issue.

I found a few problems.

1.  I've added "buysell_formatter": "{asset} {direction}" to locale-ja.json on line 1151, but it is not reflected properly on frontend. Moreover, the asset name is gone on the buy-side box.

<img width="1112" alt="not-reflected" src="https://user-images.githubusercontent.com/36555761/37534580-96db47dc-2988-11e8-974a-5a99adc6a0be.png">

2. It seems like there are a plenty of hardcoding English phrases on the deposit page. Can you make this translatable?

<img width="1211" alt="hard-coding" src="https://user-images.githubusercontent.com/36555761/37534593-a2bb0678-2988-11e8-8830-aeb99fb7ddc6.png">

3. It seems like the Borrow bitUSD screen that I have reported previously is still containing hardcoded English phrases. Can you make these translatable?

<img width="694" alt="hard-coding2" src="https://user-images.githubusercontent.com/36555761/37534641-be9d5922-2988-11e8-92e6-106cb55cc6f9.png">

**Comment by wmbutler** [comment]
Create a new issue. This one has already been paid.
